### DISCUSSION TITLE
What is Texl?

### DISCUSSION THREAD

**Discussion by bricelam** [microsoft/Power-Fx#2707] [Q&A] [answered]
I'm getting familiar with the codebase and was wondering what *Texl* means. Is it a play on T-SQL, like "T-SQL functions in Excel" shortened to "T-Excel" shortened to "Texl"?

**Comment by PowerAppsDarren** [comment]
Based on the [code found here](https://github.com/microsoft/Power-Fx/blob/main/src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs), Texl appears to be a core component of the Power Fx language implementation. The code suggests that "Texl" is an internal name for the type expression layer or type execution layer of Power Fx, though this is not explicitly documented in the source code.

Here are key insights:

## Core Functionality

### Base Class Purpose

- TexlFunction is an abstract base class that implements IFunction interface
- It provides the foundation for all function implementations in Power Fx
- Handles function signatures, parameter validation, and type checking

### Key Features

- Supports lambda parameters through a mask system
- Handles function delegation to external data sources
- Manages function metadata and localization
- Processes type coercion and validation

## Implementation Details

### Function Properties

- Name and namespace management
- Return type validation
- Parameter type checking and coercion
- Delegation capabilities
- Error handling and reporting

### Type System

- Supports complex type checking including tables and records
- Handles type coercion for parameters
- Manages column type validation
- Supports metadata type arguments

### Delegation Support

- Enables server-side execution of functions
- Handles data source queries and operations
- Manages delegation capabilities and strategies

**Comment by lesaltzm** [comment, marked as answer]
Love the suggestion of T-Excel, but the actual answer is that the language that became Power Fx started life as a project called "Tangram". TEXL showing up in the code is a leftover from those days, a short form for "Tangram EXpression Language"
